core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -983,7 +983,10 @@ template <typename Rank, typename TeamHandle, typename Lambda,
 KOKKOS_INLINE_FUNCTION void parallel_reduce(
     TeamThreadMDRange<Rank, TeamHandle> const& policy, Lambda const& lambda,
     ReducerValueType& val) {
+  val = ReducerValueType{};
```

Comment:
I have a very vague memory that we intentionally did not default initialize the result variable here, expecting it to be handled further down the line when eventually non-md team policies are called. But seeing that the result does not reflect that thought, I think initializing the result value here works.

Reply:
Do we exclusively support sum reductions?
Otherwise initializing this way does not look right to me.

Reply:
Yes, we currently only allow sum reductions for this interface.

Reply:
Do we have a static assert to detect if a non-sum reducer or if a functor with a init/join is passed?

Reply:
I added some `static_assert`s for the `value_type` in 1c85dd0b71fd1b0a2f4eb2d4a655f524e0fb4c96.
It shouldn't be that difficult to extend the interface to support the full reduction interface but I would rather do that in follow-up pull request.

Reply:
Also, note that the nested reduction interface doesn't allow functors as reducers yet, see https://github.com/kokkos/kokkos/issues/6317.